### DISCUSSION TITLE
use multiple reference for one floating component

### DISCUSSION THREAD

**Discussion by anbotz** [floating-ui/floating-ui#1886] [Q&A]
Hi,
i used floatting-ui and react-table on a project with a complexe table.
on each row, I have a button which pop a menu on click, i currently use useFloatting, it works fine
I need to stick the first column of my table, therefore i added react-table-sticky, which stick td content
but now, my dropdown menu are render behind my td, even with a high zindex

I want to render a unique dropdown menu, above my table component, which will be the unique floating of multilples reference (one at the time, obviously).

Is it possible to switch reference on click on a element ?

thanks !

**Comment by fedemp** [comment]
> my dropdown menu are render behind my td, even with a high zindex

Have you checked which parent you have set the `z-index`? Maybe the menu has a high, but its parent has a low one and that's what is used to calculate the stacking order. https://stackoverflow.com/a/32515284

> which will be the unique floating of multilples reference (one at the time, obviously).

I'm currently in a very similar situation and I was about the ask about it. Given that the reference is different each time, wouldn't it be right to give a different floating instance for each element that will trigger the floating (i.e. each reference gets its own floating). That what you (and me) would do if we were talking about tooltips, right? Moreover, giving an exclusive floating to each reference would allow to easily return focus to the trigger/reference once the floating is closed; this is given out of the box with floating-ui.
